### DISCUSSION TITLE
Difference between docker, docker-compose and budi cli

### DISCUSSION THREAD

**Discussion by renrom** [Budibase/budibase#9961] [Help] [answered]
What is the main difference between self-hosted installations between docker, docker-compose and budi cli? and why would one prefer over the other?

**Comment by melohagan** [comment, marked as answer]
Hey @renrom 

[Docker](https://docs.budibase.com/docs/docker) as described in the docs is the method for running a **single container** that holds everything needed to run Budibase.

[Docker-Compose](https://docs.budibase.com/docs/docker-compose) is simply a different method of spinning up a Budibase instance using a docker compose file. Multiple containers will be created in this case, but through the docker compose file you will be able to tweak settings for the individual services.

The [Budibase CLI](https://docs.budibase.com/docs/budibase-cli-setup) is essentially a wrapper for *docker-compose*. 
For example when you do a `budi hosting --start` that is the equivalent of `docker-compose up -d`. 

One of the benefits of using the Budibase CLI is that it makes updating your self-hosted installation easy through the `budi hosting --update` command, but also has other options for [Backups](https://docs.budibase.com/docs/budibase-cli-setup#backing-up-data-using-the-budibase-cli) and [Custom plugin development](https://docs.budibase.com/docs/custom-plugin#budibase-cli)

**Reply by renrom** [reply]
Thanks for the explanation!
